javascript/core/array-methods/shuffle-an-array-using-sort-and-random.md

```diff
@@ -35,68 +35,96 @@ links:
 Given the array:
 
 ```javascript
-var array = [1,2,3,4,5,6,7,8,9];
+let array = [1, 2, 3, 4, 5, 6, 7, 8, 9];
 ```
 
-Sort randomly with:
+You could sort it randomly using `Array.prototype.sort()`[1]:
 
 ```javascript
-array.sort(function() {
+array.sort((a, b) => {
   return Math.random() - 0.5
 });
+// The order will be different
+// every time
+// [5, 4, 2, 1, 6, 3, 7, 8, 9]
 ```
 
-The `- 0.5` means the returned value exists between the range `-0.5 < x < 0.5`, allowing the sort function to decide where to position the next item.
+This is due to how the `Array.prototype.sort()` method works. The arrow function we pass as an argument will return a number, and depending on its value it will sort the array:
+- 0 -> `a = b`
+- > 0 -> `a > b`
+- < 0 -> `b > a`
 
-An example result:
-
-```javascript
-array = [4,3,1,2,9,5,6,8,7]
-```
+In our example, we return `Math.random() - 0.5` which works like this:
+- `Math.random()` return a random number between 0 and 1 (`0 < Math.random() < 1`)
+- we subtract `0.5` from the random number (`-0.5 < Math.random() - 0.5 < 0.5`)
+- we sort the array depending on the result
 
 ---
 ## Practice
 
 Complete the code such that it will shuffle an array using `sort` and `random`.
 
 ```javascript
-var arr = [3, 1, 4, 2, 7, 1]
-arr.???(???() {
+let arr = [3, 1, 4, 2, 7, 1]
+arr.???((???) => {
   return ???.???() - 0.5
 });
 
 // arr = [4, 2, 7, 1, 1, 3]
 ```
 
 * sort
-* function
+* a, b
 * Math
 * random
 * shuffle
-* () =>
+* x
 * arr
-* merge
 * prototype
 * Array
 * call
 * apply
-* +=
-* -0.5
 
 ---
 ## Revision
 
 The following code will ???
 
 ```javascript
-var arr = [1,2,3,4,5,6,7];
-arr.sort(function(){
+let arr = [1, 2, 3, 4, 5, 6, 7];
+arr.sort(() => {
   return Math.random() - 0.5;
 });
 ```
 
 * shuffle the array
 * sort the array decreasingly
 * do nothing to the array
- 
- 
+
+---
+## Footnotes
+
+[1:Array.prototype.sort()]
+This method, when used on its own, converts the elements of the given array into strings, and sorts them by comparing their sequences of UTF-16 code units value:
+```js
+let days = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday'];
+days.sort();
+console.log(days);
+// Output:
+// ['Friday', 'Monday', 'Thursday', 'Tuesday', 'Wednesday']
+```
+
+You can also pass a function as an argument. The function can take `(a, b)` as input, representing a pair of 2 elements in the array that the function will be called on. It must return a number, and depending on the returned number it means:
+- 0 -> `a = b`
+- > 0 -> `a > b`
+- < 0 => `a < b`
```

Comment:
If we we're to apply my comments from above, this part would need a bit of adjustment too